sktime/forecasting/ttm.py

```diff
@@ -288,11 +310,27 @@ def _fit(self, y, X, fh):
                 TinyTimeMixerForPrediction,
             )
 
-        # Get the Configuration
-        config = TinyTimeMixerConfig.from_pretrained(
-            self.model_path,
-            revision=self.revision,
-        )
+        # Initialize to self.config, then adjust accordingly
+        config = self.config
+        if self.model_path is None:
+            if self.fit_strategy != "full":
+                raise ValueError(
+                    "Invalid configuration: 'model_path' is set to None."
+                    "This requires 'fit_strategy' to be 'full'."
+                    "Please set 'fit_strategy' to 'full' or provide a valid model path."
+                )
+            if config is None:
+                # Load tinytimemixer config
+                config = TinyTimeMixerConfig()
+                config.check_and_init_preprocessing()
+            else:
+                config = TinyTimeMixerConfig(**config)
+        else:
+            # Get the pretrained model Configuration
+            config = TinyTimeMixerConfig.from_pretrained(
+                self.model_path,
+                revision=self.revision,
+            )
```

Comment:
this code can be simplified as below, right? since the config is updated with the user provided config in the later code
```suggestion
        if self.model_path is None:
            config = TinyTimeMixerConfig()
        else:
            config = TinyTimeMixerConfig.from_pretrained(
                self.model_path,
                revision=self.revision,
            )
```

Reply:
Actually the code
```
        else:
            # Initialize model with default config
            self.model = TinyTimeMixerForPrediction(config=config)
```
requires config to be the object of TinyTimeMixerConfig Class.

Reply:
I still don't understand, the object is always of `TinyTimeMixerConfig` from any given condition

Reply:
But if we don't do `config = TinyTimeMixerConfig(**config)` in case of model_path and config being None, then config will be updated with user provided config which is a dict and hence will raise an Error.

Reply:
you won't need to pass `**config` to `TinyTimeMixerConfig`, just load the default one without giving any config, this config is updated in the later code